library/src/scala/collection/mutable/ListBuffer.scala

```diff
@@ -188,7 +189,7 @@ class ListBuffer[A]
       last0 = null
   }
 
-  private def locate(i: Int): Predecessor[A] =
+  private def locate(i: Int): Predecessor[A] | Null =
```

Comment:
Why is this necessary when `Predecessor` is defined as nullable above (in `private type Predecessor = ...`)?